db/migrate/20210401144737_set_first_and_second_course_years_for_core_induction_programme.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class SetFirstAndSecondCourseYearsForCoreInductionProgramme < ActiveRecord::Migration[6.1]
+  def up
+    CourseYear.all.each do |course_year|
+      cip = CoreInductionProgramme.find(course_year[:core_induction_programme_id])
+      if course_year.is_year_one?
+        cip.update!(course_year_one_id: course_year[:id])
+      else cip.update!(course_year_two_id: course_year[:id])
+      end
+    end
+  end
+
+  def down
+    CoreInductionProgramme.all.each do |cip|
+      course_year = CourseYear.where(cip[:course_year_one_id]).or(CourseYear.where(cip[:course_year_two_id])).first
+      if cip[:course_year_one_id].nil?
+        course_year.update!(is_year_one: false)
+      else
+        course_year.update!(is_year_one: true)
+      end
+    end
```

Comment:
This is incomplete too. You need to consider the case when both years are present :)